crates/valence_boss_bar/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+#![allow(clippy::type_complexity)]
+
+use bevy_app::Plugin;
+use bevy_ecs::query::{Added, Changed};
+use bevy_ecs::removal_detection::RemovedComponents;
+use bevy_ecs::system::Query;
+use components::{BossBarFlags, BossBarHealth, BossBarStyle, BossBarTitle, BossBarViewers};
+use packet::{BossBarAction, BossBarS2c};
+use valence_client::Client;
+use valence_core::despawn::Despawned;
+use valence_core::protocol::encode::WritePacket;
+use valence_core::uuid::UniqueId;
+
+pub mod components;
```

Comment:
Consider making this module private and re-exporting everything publicly, since the only public item in the root of the crate is `BossBarPlugin`.

```suggestion
mod components;
pub use components::*;
```